test/API/H5_api_test_util.c

```diff
@@ -729,37 +721,22 @@ prefix_filename(const char *prefix, const char *filename, char **filename_out)
 }
 
 /*
- * Calls H5Fdelete on the given filename. If a prefix string
- * is given, adds that prefix string to the filename before
- * calling H5Fdelete
+ * Calls H5Fdelete on the given filename as long as cleanup
+ * of testing files has not been disabled.
  */
 herr_t
-remove_test_file(const char *prefix, const char *filename)
+remove_test_file(const char *filename)
 {
-    const char *test_file;
-    char       *prefixed_filename = NULL;
-    herr_t      ret_value         = SUCCEED;
+    herr_t ret_value = SUCCEED;
 
     if (!GetTestCleanup())
         goto done;
 
-    if (prefix) {
```

Comment:
The prefix argument was NULL in every single place this function was used and needed to be removed after refactoring.